static/app/views/explore/tables/aggregatesTable.tsx

```diff
@@ -69,13 +70,19 @@ export function AggregatesTable({
   const groupBys = useExploreGroupBys();
   const visualizes = useExploreVisualizes();
 
-  const {result, eventView, fields} = aggregatesTableResult;
+  const {result, fields} = aggregatesTableResult;
 
   const sorts = useExploreSortBys();
   const setSorts = useSetExploreSortBys();
   const query = useExploreQuery();
 
-  const columns = useMemo(() => eventView.getColumns(), [eventView]);
+  const columns = useMemo(() => {
+    return decodeColumnOrder(
+      fields.map(f => {
+        return {field: f};
+      })
+    );
+  }, [fields]);
```

Comment:
Looking at this more closely, I think your previous solution was better because the control over what `columns` will be is in 1 place. Though 1 suggestion I think that would make this more robust is for the hooks to directly return `columns` instead of the `eventView`. That way, we can guarantee that the indices of `fields` and `columns` correspond to the same values all the time.

The same applies for both the samples and aggregates query.